tests/test_task_loader.py

```diff
@@ -270,6 +270,103 @@ def test_links(self) -> None:
             tl = TaskLoader(context=self.df, target=self.df, links=[(0, 0)])
             task = tl("2020-01-01", "gapfill", "gapfill")
 
+    @parameterized.expand([[(0.3, 0.3)], [(0.6, 0.4)]])
+    def test_patch_size(self, patch_size) -> None:
+        """Test patch size sampling."""
+        # need to redefine the data generators because the patch size samplin
+        # where we want to test that context and or target have different
+        # spatial extents
+        da_data_0_1 = self.da
+
+        # smaller normalized coord
+        da_data_smaller = _gen_data_xr(
+            coords=dict(
+                time=pd.date_range("2020-01-01", "2020-01-31", freq="D"),
+                x1=np.linspace(0.1, 0.9, 25),
+                x2=np.linspace(0.1, 0.9, 10),
+            )
+        )
+        # larger normalized coord
+        da_data_larger = _gen_data_xr(
+            coords=dict(
+                time=pd.date_range("2020-01-01", "2020-01-31", freq="D"),
+                x1=np.linspace(-0.1, 1.1, 50),
+                x2=np.linspace(-0.1, 1.1, 50),
+            )
+        )
+
+        context = [da_data_0_1, da_data_smaller, da_data_larger]
+        tl = TaskLoader(
+            context=context,  # gridded xarray and off-grid pandas contexts
+            target=self.df,  # off-grid pandas targets
+        )
+
+        # TODO it would be better to do this with pytest.fixtures
+        # but could not get to work so far
+        task = tl(
```

Comment:
It would be better to have fixtures that generate the data setup and then we can test with different configurations like single date, list of dates, different context and target sampling strategies  etc.